### DISCUSSION TITLE
[TPI] Duda sobre el enunciado

### DISCUSSION THREAD

**Discussion by Darkonis13** [dds-jv/foro#251] [Consultas generales] [answered]
Buenas, quería consultar qué significan las siglas 0x0, 0x10, 0x0 que aparecen en el enunciado de FS...

> _Entonces, tenemos como tarea "embellecer" al sistema de archivos. Y como segunda tarea, dar un ejemplo de uso del API para los siguiente casos:_
> 
> _Tenemos que leer de un archivo 3 campos: el primero de 4 bytes (C0), el segundo de 1 byte (C1), el tercero de 5 bytes (C2). Y escribir en otro archivo C0, un bloque 0x0, 0x10, 0x0, y luego C1 y C2.
> Tenemos que leer un archivo completo, y escribirlo en otro archivo, en bloques de igual tamaño parametrizable.
> Los ejemplos tambien tenemos que plantearlos nosotres, en forma de tests._
> 
> _Finalmente, como nuestro cliente es bastante quisquilloso quiere ver formas alternativas de solucionar las lecturas sincrónicas y asincrónicas, para compararlas y ver cuales le gustan más_

No me queda muy claro cuál es el requisito a satisfacer 🥺 muchas gracias

**Comment by nanderson87** [comment, marked as answer]
Hola @Darkonis13 

0x básicamente es una representación de una número hexadecimal. Por ejemplo 0x0 representa el hexadecimal 0 (decimal 0) y 0x10 el hexadecimal 10 (decimal 16).

Respecto al requerimiento sería utilizar la nueva interfaz que creamos para cumplir con estos dos ejemplos (a través de test). 
- Leer tres campos de un archivo y escribirlo en otro con las condición que nos plantean respecto al orden y los bytes de cada campos .
- Leer un archivo completo y escribirlo en otro donde el tamaño de lectura y escritura sea parametrizable. 

En resumen, son dos casos que nos piden desarrollar a modo de ejemplo para ver como se harían a partir de haber desarrollado la nueva interfaz.
